### PR TITLE
Optional plugins: move documentation to plugin themsevels

### PR DESCRIPTION
**Pull request by clebergnu**
The documentation for the optional plugins are currently hosted in the main Avocado "directory".  It's certainly not the best location
for many reasons:
    
* The documentation is not available the Python package (such as when browsing pypi)
* The documentation is not available when browsing the sources on Git forges (such as GitHub).
* Developers hacking the code won't easily find the documentation.

This moves the documentation to the plugin directory itself, while still keeping it in the "Plugins" documentation section.

### CODE REVIEW

File: optional_plugins/html/MANIFEST.in [hunk 1]
```diff
@@ -1,2 +1,2 @@
 recursive-include avocado_result_html/templates *
-include VERSION
\ No newline at end of file
+include VERSION README.rst
```
Comment (richtja) [inline optional_plugins/html/MANIFEST.in:2]: new line missing

File: optional_plugins/golang/setup.py [hunk 2]
```diff
@@ -23,12 +25,22 @@
...
+        readme_contents = readme.read()
+    return readme_contents
+
 
 setup(
     name="avocado-framework-plugin-golang",
-    description="Avocado Plugin for Execution of Golang tests",
     version=VERSION,
+    description="Avocado Plugin for Execution of Golang tests",
+    long_description=get_long_description(),
```
Comment (richtja) [inline optional_plugins/golang/setup.py:43]: If we are adding `long_description` IMO, based on [documentation](https://packaging.python.org/en/latest/guides/making-a-pypi-friendly-readme/), we should also add `long_description_content_type` attributes. This is related to all commits from this PR.

File: optional_plugins/golang/MANIFEST.in [hunk 3]
```diff
@@ -1 +1 @@
-include VERSION
+include VERSION README.rst
```
Comment (richtja) [inline optional_plugins/golang/MANIFEST.in:1]: You added a very good description of why this change is needed in this commit, but I am missing the same thing in the other commits. Can you please copy/paste this to the other commits. IMO, it will increase commit history readability.

**Review by richtja — CHANGES_REQUESTED** [review]
Hi @clebergnu, thanks for these changes, it will definitely increase readability of our documentation, well done. I have just a few style comments to consider.

File: optional_plugins/html/MANIFEST.in (hunk 1 above)
Comment (clebergnu) [reply optional_plugins/html/MANIFEST.in:2]: Actually, this is how all the `MANIFEST.in` are currently.  But, sure, I can add newlines.

**Comment by clebergnu** [comment]
@richtja given the changes you requested are minor, I've pushed the changes into the same branch/PR.

Thanks for the review!

**Review by richtja — APPROVED** [review]
LGTM, thank you.
